### PR TITLE
Added Settings (user preferences) menu

### PR DESCRIPTION
**Pull request by flabbet**
Added Settings menu in `File -> Settings` with settings:
- Default new file size
- Check updates on startup
- Show New File dialog on startup

Of course, added `user_preferences.json` reader and writer. The file is stored under `AppData\Local\PixiEditor\user_preferences.json`

### CODE REVIEW

File: PixiEditor/Views/MainWindow.xaml [hunk 1]
```diff
@@ -83,6 +83,8 @@
                               Command="{Binding FileSubViewModel.SaveDocumentCommand}" CommandParameter="AsNew" />
                     <MenuItem Header="_Export" InputGestureText="Ctrl+Shift+Alt+S" Command="{Binding FileSubViewModel.ExportFileCommand}" />
                     <Separator />
+                    <MenuItem Header="_Settings" Command="{Binding MiscSubViewModel.OpenSettingsWindowCommand}" />
```
Comment (CPKreuz) [inline PixiEditor/Views/MainWindow.xaml:86]: I would expect it to be in the `Edit` tab

File: PixiEditor/Views/Dialogs/SettingsWindow.xaml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        <WindowChrome CaptionHeight="32"
+                      ResizeBorderThickness="{x:Static SystemParameters.WindowResizeBorderThickness}" />
+    </WindowChrome.WindowChrome>
+
+    <Window.CommandBindings>
+        <CommandBinding Command="{x:Static SystemCommands.CloseWindowCommand}" CanExecute="CommandBinding_CanExecute"
+                        Executed="CommandBinding_Executed_Close" />
+    </Window.CommandBindings>
+
+    <Grid Background="{StaticResource AccentColor}">
```
Comment (CPKreuz) [inline PixiEditor/Views/Dialogs/SettingsWindow.xaml:25]: Might look better if you switch the colors

```suggestion
    <Grid Background="{StaticResource MainColor}">
```

File: PixiEditor/Views/Dialogs/SettingsWindow.xaml [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+                    WindowChrome.IsHitTestVisibleInChrome="True" ToolTip="Close"
+                    Command="{x:Static SystemCommands.CloseWindowCommand}" />
+        </DockPanel>
+        <StackPanel Grid.Row="1" Grid.Column="0">
+            <Button Style="{StaticResource DarkRoundButton}" Margin="10 5 10 5"
+                    Command="{Binding SelectCategoryCommand}" CommandParameter="General">General</Button>
+            <Button Style="{StaticResource DarkRoundButton}" Margin="10 5 10 5" 
+                    Command="{Binding SelectCategoryCommand}" CommandParameter="Updates">Updates</Button>
+        </StackPanel>
+        <Grid Grid.Row="1" Grid.Column="1" Background="{StaticResource MainColor}">
```
Comment (CPKreuz) [inline PixiEditor/Views/Dialogs/SettingsWindow.xaml:47]:
```suggestion
        <Grid Grid.Row="1" Grid.Column="1" Background="{StaticResource AccentColor}">
```

File: PixiEditor/Views/MainWindow.xaml (hunk 1 above)
Comment (flabbet) [reply PixiEditor/Views/MainWindow.xaml:86]: I was thinking about that, some software has this in `File` some in `Edit`. Not sure which one should we use.

Comment (flabbet) [reply PixiEditor/Views/MainWindow.xaml]: Ok, I moved it to `Edit`

**Review by CPKreuz — APPROVED** [review]

**Pull request merged**